### DISCUSSION TITLE
Pylance should treat `# %%` blocks within a `.py` file as notebook cells

### DISCUSSION THREAD

**Discussion by Marmeladenbrot** [microsoft/pylance-release#5322] [Enhancement]
Regarding the closed issue [https://github.com/microsoft/vscode-jupyter/issues/1497](https://github.com/microsoft/vscode-jupyter/issues/1497):

`%%timeit ` in a notebook cell in a .py file does not work it you replace it with `#! %%timeit` as proposed as the solution in the closed ticket.

Works but Pylance underscores the first % in %%timeit :
```
# %%
%%timeit
sum([1, 2, 3])
# %%
```

Works not:
```
# %%
#! %%timeit
sum([1, 2, 3])
# %%

```
Please reopen this issue

**Comment by debonte** [comment]
Moving this issue to discussion as an enhancement request for comments and upvotes. Pylance currently doesn't recognize the `# %%` syntax for declaring cell blocks within a .py file. If you're interested in seeing this enhancement implemented, please up vote that the discussion.
